js/components/frame.js

```diff
@@ -224,29 +231,23 @@ class Frame extends ImmutableComponent {
     this.webview.addEventListener('did-fail-load', () => {
     })
     this.webview.addEventListener('did-finish-load', () => {
-    })
-    this.webview.addEventListener('did-navigate-in-page', () => {
       WindowActions.onWebviewLoadEnd(
         this.props.frame,
         this.webview.getURL())
       this.webview.send(messages.POST_PAGE_LOAD_RUN)
+      let security = this.props.frame.get('security')
+      if (this.props.frame.get('location') === 'about:certerror' &&
+          security && security.get('certDetails')) {
+        // Don't send certDetails.cert since it is big and crashes the page
+        this.webview.send(messages.CERT_DETAILS_UPDATED, {
+          url: security.get('certDetails').url,
+          error: security.get('certDetails').error
+        })
+      }
+    })
+    this.webview.addEventListener('did-navigate-in-page', () => {
```

Comment:
did-finish-load is only called for the main frame
`void WebContents::DidFinishLoad(content::RenderFrameHost* render_frame_host,
                                const GURL& validated_url) {
  bool is_main_frame = !render_frame_host->GetParent();
  Emit("did-frame-finish-load", is_main_frame);

  if (is_main_frame)
    Emit("did-finish-load");
}
`